### PR TITLE
[Feature] Program Execute Command

### PR DESCRIPTION
**Pull request by ilitteri**
## Motivation

This PR addresses #356

### CODE REVIEW

File: cli/commands/execute.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        // Derive the view key from the private key
+        let view_key = ViewKey::try_from(private_key)?;
+
+        // TODO: Find a better way to filter commitments from the program execution inputs.
+        let mut inputs: Vec<Value<Network>> = Vec::new();
+        self.inputs.into_iter().try_for_each(|input| {
+            if input.to_string().ends_with("field") {
+                let ciphertext: Record<Network, Ciphertext<Network>> =
+                    ureq::get(&format!("http://localhost/testnet3/ciphertext/{input}"))
```
Comment (howardwu) [inline cli/commands/execute.rs]: Can we change these URLs from `localhost` to an opinionated hosted endpoint?

**Comment by howardwu** [comment]
This PR needs rebasing

File: cli/commands/execute.rs (hunk 1 above)
Comment (ilitteri) [reply cli/commands/execute.rs]: #338 fixes this issue for deployments when it's merged I'll make the same changes for executions.

File: cli/commands/execute.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                let record = ciphertext.decrypt(&view_key)?;
+                inputs.push(Value::Record(record));
+            } else {
+                inputs.push(input);
+            }
+            Ok::<(), Error>(())
+        })?;
+
+        // Find an unspent ciphertext to use later for the fee.
+        let ciphertext = ureq::post("http://localhost/testnet3/ciphertexts/unspent")
```
Comment (evanmarshall) [inline cli/commands/execute.rs:75]: I think this should be: `/records/unspent`. Should it also be more careful to ensure that we're spending the correct records? Like if we have a record representing other data, we may not necessarily want to spend it for this transaction. We'd want to find a record with no other data fields.

**Pull request closed without merging**